src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -231,6 +238,20 @@ protected async Task<IActionResult> ExecuteQuery<TPostBodyResource>(TResource re
                     List<DataProviderMetadata> dataProvidersMetadata = null;
                     Response invocationResponse = null;
                     bool isInternalCall = true;
+                    QueryUtterancesResults utterancesResults = null;
+
+                    // Get suggested utterances for the detector
+                    string[] utterances = null;
+                    if (jsonBody.DetectorUtterances != null && invoker.EntryPointDefinitionAttribute.Description.ToString().Length > 3)
+                    {
+                        utterances = JsonConvert.DeserializeObject<string[]>(jsonBody.DetectorUtterances);
+                        string description = invoker.EntryPointDefinitionAttribute.Description.ToString();
+                        var resourceParams = _internalApiHelper.GetResourceParams(invoker.ResourceFilter);
+                        var searchUtterances = await _searchService.SearchUtterances(runtimeContext.OperationContext.RequestId, description, utterances, resourceParams);
+                        string resultContent = await searchUtterances.Content.ReadAsStringAsync();
+                        utterancesResults = JsonConvert.DeserializeObject<QueryUtterancesResults>(resultContent);
```

Comment:
This  casting will fail in case of response other than success  from Search API with JSON Deserialization error.

Should you include Success check here?

Reply:
Done.